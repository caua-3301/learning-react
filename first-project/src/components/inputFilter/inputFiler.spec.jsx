import { render, screen } from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import { MyInput } from '.';

describe('<MyInput/>', () => {
  it('should write', () => {
    expect.assertions(3);

    const fn = jest.fn();

    render(<MyInput hadleChange={fn} />);

    const inputText = screen.getByPlaceholderText('Digite aqui');

    const input = 'PesquisaTeste';

    userEvent.type(inputText, input);

    expect(inputText.value).toBe(input);
    expect(fn).toHaveBeenCalledTimes(input.length);
    expect(inputText).toBeInTheDocument();
  });

  it('make the snapshot', () => {
    const fn = jest.fn();

    render(<MyInput hadleChange={fn} />);
    const elementInput = screen.getByPlaceholderText('Digite aqui');

    expect(elementInput).toMatchSnapshot();
  });
});
