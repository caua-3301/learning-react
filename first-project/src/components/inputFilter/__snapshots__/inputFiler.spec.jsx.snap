// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyInput/> make the snapshot 1`] = `
<input
  placeholder="Digite aqui"
  type="search"
  value=""
/>
`;
